### PR TITLE
Refactoring: Rename LAST_SYNCED_L1_BLOCK to be more clear

### PR DESCRIPTION
**Pull request by dangush**
# Refactor

Closes Issue #1335 by refactoring LAST_SYNCED_L1_BLOCK to a struct named LastSyncedEventBlock which contains a block_number field and event_index field. 

## What is the current behavior?

When `process_l1_message` is called, the L1 block number is written to the MessagingDb. 

Resolves: #1335 

## What is the new behavior?

When `process_l1_message` is called, a struct containing both the L1 block number and event_index is written to the MessagingDb. 

## Does this introduce a breaking change?

I'm not yet sure if/how the MessagingDb schema needs to be changed to accept the encoded Struct instead of previously-encoded u64 value. Accorrding to the [SCALE encoding docs](https://docs.substrate.io/reference/scale-codec/) I believe structs get encoded as a vector of bytes, so it seems the difference is that a value twice the current size will need to be written to the DB.

## Todo

I would like to write tests for this part of the code, and have asked a few questions in the issue thread asking how to approach this.

### CODE REVIEW

**Review by apoorvsadana — APPROVED** [review]
lgtm

**Comment by apoorvsadana** [comment]
@tdelabro can you review this once?

File: crates/client/db/src/lib.rs [hunk 1]
```diff
@@ -19,7 +19,7 @@ pub use mapping_db::MappingCommitment;
 use starknet_api::hash::StarkHash;
 mod da_db;
 mod db_opening_utils;
-mod messaging_db;
+pub mod messaging_db;
```
Comment (tdelabro) [inline crates/client/db/src/lib.rs:22]: We don't want any of the db `mod` to be public. They should only be accessible through our `Backend` struct

Comment (tdelabro) [reply crates/client/db/src/lib.rs:22]: I would reexport `LastSyncedEventBlock` struct in `lib.rs`

**Review by tdelabro — CHANGES_REQUESTED** [review]

File: crates/client/db/src/messaging_db.rs [hunk 2]
```diff
@@ -14,18 +14,37 @@ pub struct MessagingDb<B: BlockT> {
...
+        LastSyncedEventBlock { block_number, event_index }
+    }
+}
+
 impl<B: BlockT> MessagingDb<B> {
-    pub fn last_synced_l1_block(&self) -> Result<u64, DbError> {
-        match self.db.get(crate::columns::MESSAGING, crate::static_keys::LAST_SYNCED_L1_BLOCK) {
-            Some(raw) => Ok(u64::decode(&mut &raw[..])?),
-            None => Ok(0),
+    pub fn last_synced_l1_block_with_event(&self) -> Result<LastSyncedEventBlock, DbError> {
```
Comment (tdelabro) [inline crates/client/db/src/messaging_db.rs:30]: `last_synced_l1_block_and_event`

**Review by tdelabro — APPROVED** [review]

**Pull request merged**